### PR TITLE
Add REST API to update user profiles.

### PR DESCRIPTION
**Pull request by SeraVault**
This functionality allows the end client to call /change-profile that will allow updating the user's profile.

### CODE REVIEW

File: src/user.ts [hunk 1]
```diff
@@ -1104,7 +1113,7 @@ export class User {
       req.user.provider,
       user
     );
-    await this.userDB.insert(finalUser);
```
Comment (LyteFM) [inline src/user.ts:1107]: This completely unrelated change would introduce a security vulnerability.

**Review by LyteFM — COMMENTED** [review]
See my comment https://github.com/perfood/couch-auth/issues/81#issuecomment-1873976544. I'm rather conservative about adding new features that I don't intend to use myself.

File: src/user.ts (hunk 1 above)
Comment (SeraVault) [reply src/user.ts:1107]: Okay.  But I didn't see a way to verify that the session belonged to the requestor user_id.  I'll look at your code again so that I can implement my own very secure implementation.  
Profile information is implemented via Oauth, right?  Why would that not be requisite functionality for local accounts?

Comment (SeraVault) [reply src/user.ts:1107]: private async completeEmailChange doesn't seem to validate that the session is owned by the requestor.  Theoretically, I could initiate a session and change anyone's email.  How is this handled internally to prevent spoofing?  That's what I'm not seeing.
